lib/dataset/test/bin_test.cpp

```diff
@@ -178,6 +178,26 @@ TEST_P(BinTest, rebin_no_event_coord) {
   EXPECT_THROW_DISCARD(bin(x, {edges_x}), except::BinEdgeError);
 }
 
+TEST_P(BinTest, bin_using_attr) {
+  auto table = GetParam();
+  const auto expected = bin(table, {edges_x});
+  table.attrs().set(Dim::X, table.coords().extract(Dim::X));
+  const auto result = bin(table, {edges_x});
+  auto view = bins_view<DataArray>(result.data());
+  view.coords().set(Dim::X, view.attrs().extract(Dim::X));
+  EXPECT_EQ(expected, result);
+}
+
+TEST_P(BinTest, rebin_using_attr) {
+  auto table = GetParam();
+  const auto expected = bin(bin(table, {edges_x}), {edges_x_coarse});
+  table.attrs().set(Dim::X, table.coords().extract(Dim::X));
+  const auto result = bin(bin(table, {edges_x}), {edges_x_coarse});
+  auto view = bins_view<DataArray>(result.data());
+  view.coords().set(Dim::X, view.attrs().extract(Dim::X));
+  EXPECT_EQ(expected, result);
+}
+
```

Comment:
Is this change to `bin` (and the tests) omitting the case of inputs that have bin attrs? Here you tested rebinning with event attrs, but what about rebinning with bin attrs, defining existing binning?

Reply:
Do you mean binning something that looks like this?
![Screenshot at 2022-01-21 12-59-54](https://user-images.githubusercontent.com/39047984/150523719-3d527fc2-d890-4608-8d2f-fb45948a39e0.png)

In this case, after the second binning, should the wrapper DataArray have `x` as a coord or an attr?

Reply:
> In this case, after the second binning, should the wrapper DataArray have x as a coord or an attr?

Newly specified edges should always be coords.

Reply:
agreed